plugins/c8y_configuration_plugin/src/main.rs

```diff
@@ -134,65 +135,65 @@ async fn run(
-                        // retrieve config file upload smartrest request from payload
-                        let config_upload_request =
-                            SmartRestConfigUploadRequest::from_smartrest(payload)?;
-
-                        // handle the config file upload request
-                        handle_config_upload_request(
-                            &plugin_config,
-                            config_upload_request,
-                            &mut mqtt_client,
-                            http_client,
-                        )
-                        .await
-                    }
-                    _ => {
-                        // Ignore operation messages not meant for this plugin
-                        Ok(())
+    let mut inotify_stream = inofity_file_watch_stream(config_file_path, WatchMask::CLOSE_WRITE)?;
+
+    loop {
+        tokio::select! {
+            message = mqtt_client.received.next() => {
+                if let Some(message) = message {
+                    debug!("Received {:?}", message);
+                    if let Ok(payload) = message.payload_str() {
+                            match payload.split(',').next().unwrap_or_default() {
+                                "524" => {
+                                    let config_download_request =
+                                        SmartRestConfigDownloadRequest::from_smartrest(payload)?;
+
+                                    handle_config_download_request(
+                                        &plugin_config,
+                                        config_download_request,
+                                        tmp_dir.clone(),
+                                        &mut mqtt_client,
+                                        http_client,
+                                    )
+                                    .await?
+                                }
+                                "526" => {
+                                    // retrieve config file upload smartrest request from payload
+                                    let config_upload_request =
+                                        SmartRestConfigUploadRequest::from_smartrest(payload)?;
+
+                                    // handle the config file upload request
+                                    handle_config_upload_request(
+                                        &plugin_config,
+                                        config_upload_request,
+                                        &mut mqtt_client,
+                                        http_client,
+                                    )
+                                    .await?
+                                }
+                                _ => {
+                                    // Ignore operation messages not meant for this plugin
+                                }
+                            }
+                        }
+                    } else {
+                        return Ok(());
```

Comment:
By returning here, the plugin stops when receiving a non-utf8 message. An error should be displayed and the program must continue.

Reply:
This won't show as outdated but I made some, please check @rina23q

Reply:
I still don't get it why here should return `Ok(())`?

Reply:
If the mqtt connection is closed it will return `None`. The `else` block will pick it up and close the main function
```rust
if let Some(message) = message {
   ...
} else {
  // message is None and the connection has been closed
  return Ok(());
}

```